crates/worker/src/partition/state_machine/mod.rs

```diff
@@ -582,8 +582,16 @@ impl<Codec: RawEntryCodec> StateMachine<Codec> {
                     }
                 }
             }
-            InvocationStatus::Killed(_) => {
-                todo!()
+            InvocationStatus::Killed(metadata) => {
+                self.send_response_to_sinks(
+                    ctx,
+                    service_invocation.response_sink.take().into_iter(),
+                    KILLED_INVOCATION_ERROR,
+                    Some(invocation_id),
+                    None,
+                    Some(&metadata.invocation_target),
+                )
+                .await?;
```

Comment:
Same question here regarding appending a new response sink instead of adding more special cases.

Reply:
Same comment as above, I think it's ok as long as it's explicit.